src/overmap.cpp

```diff
@@ -3332,6 +3426,72 @@ void overmap::place_cities()
     }
 }
 
+void overmap::place_stations()
+{
+    int op_station_size = get_world_option<int>( "CITY_SIZE" );
+    if( op_station_size <= 0 ) {
+        return;
+    }
+    int op_station_spacing = get_world_option<int>( "CITY_SPACING" );
+
+    // spacing dictates how much of the map is covered in stations
+    //   station  |  stations  |   size N stations per overmap
+    // spacing | % of map |  2  |  4  |  8  |  12 |  16
+    //     0   |   ~99    |2025 | 506 | 126 |  56 |  31
+    //     1   |    50    |1012 | 253 |  63 |  28 |  15
+    //     2   |    25    | 506 | 126 |  31 |  14 |   7
+    //     3   |    12    | 253 |  63 |  15 |   7 |   3
+    //     4   |     6    | 126 |  31 |   7 |   3 |   1
+    //     5   |     3    |  63 |  15 |   3 |   1 |   0
+    //     6   |     1    |  31 |   7 |   1 |   0 |   0
+    //     7   |     0    |  15 |   3 |   0 |   0 |   0
+    //     8   |     0    |   7 |   1 |   0 |   0 |   0
+
+    const double omts_per_overmap = OMAPX * OMAPY;
+    const double station_map_coverage_ratio = 1.0/std::pow(2.0, op_station_spacing);
+    const double omts_per_station = (op_station_size*2+1) * (op_station_size*2+1) * 3 / 4;
+
+    // how many stations on this overmap?
+    const int NUM_STATIONS =
+        roll_remainder(omts_per_overmap * station_map_coverage_ratio / omts_per_station);
+
+    // place a seed for NUM_STATIONS stations, and maybe one more
+    while ( stations.size() < size_t(NUM_STATIONS) ) {
+        // randomly make some stations smaller or larger
+        int size = rng(op_station_size-1, op_station_size+1);
+        if(one_in(3)) {          // 33% tiny
+            size = 1;
+        } else if (one_in(2)) {  // 33% small
+            size = size * 2 / 3;
+        } else if (one_in(2)) {  // 17% large
+            size = size * 3 / 2;
+        } else {                 // 17% huge
+            size = size * 2;
+        }
+        size = std::max(size,1);
+
+        // TODO put stations closer to the edge when they can span overmaps
+        // don't draw stations across the edge of the map, they will get clipped
+        int cx = rng(size - 1, OMAPX - size);
+        int cy = rng(size - 1, OMAPY - size);
+        if (ter(cx, cy, 0) == settings.default_oter ) {
+            ter(cx, cy, 0) = oter_id( "railroad_nesw" ); // every station starts with an intersection
+            city tmp;
+            tmp.x = cx;
+            tmp.y = cy;
+            tmp.s = size;
+            stations.push_back(tmp);
+
+            const auto start_dir = om_direction::random();
+            auto cur_dir = start_dir;
+
+            do {
+                build_city_street( cx, cy, size, cur_dir, tmp );
+            } while( ( cur_dir = om_direction::turn_right( cur_dir ) ) != start_dir );
+        }
+    }
+}
```

Comment:
This looks like a giant copy of an existing function - place_cities or something similar. That's a very bad sign.